src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.service.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User user = userService.findByEmail(email).orElse(new User());
+        if (user.getEmail() != null) {
+            boolean userExist = user.getEmail().equals(email);
+            boolean passwordIsCorrect = HashUtil.hashPassword(password, user.getSalt())
+                    .equals(user.getPassword());
+
+            if (userExist && passwordIsCorrect) {
+                System.out.println("User: " + user.getEmail() + " was successfully logged in.");
+                return user;
+            }
+        }
+        throw new AuthenticationException("Can`t log in user: "
+                + user.getEmail() + ". Login or password are incorrect");
```

Comment:
In the `login` method, if the user is not found, the exception message will include `null` for the email. Consider using the provided email parameter in the exception message instead of `user.getEmail()` to provide more accurate feedback.